src/main/java/com/puppycrawl/tools/checkstyle/api/ViolationDispatcher.java

```diff
@@ -24,7 +24,7 @@
 /**
  * Used by FileSetChecks to distribute AuditEvents to AuditListeners.
  */
-public interface MessageDispatcher {
+public interface ViolationDispatcher {
```

Comment:
This by fact, new class.
This good opportunity to make it with better design.

Major concerns on `void fireErrors(String fileName, SortedSet<Violation> errors);`
Name of method is not good. Should be `fireViolations` or similar.

@rdiachenko , what would you recommend?
We can change all.

Reply:
```
public interface ViolationDispatcher{
  void fireFileStarted(String fileName);
  void fireFileFinished(String fileName);
  void fireErrors(String fileName, SortedSet<Violation> errors);
}
```

Name of class and method becomes weird. Before update it was also not good, but it was too abstract it was less a concern, `MessageDispatcher`.

Problem is that we mixing `fireFileXxxx` with `fireViolation` and call it ViolationDispatrer, that is good name if only `fireViolation` be in interface.

May be we should just rename method in this interface to `fireViolation` and name of class `EventDispatcher`. Do we have any EventXxxx in our code base?

Reply:
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/AuditEvent.java

Reply:
> We can not use Event word.

@romani , this PR is already outdated by 2 weeks. So, how do you suggest we finish this?

Reply:
@rdiachenko , is there any suggestion on api ?
@nrmancuso , @rnveach , please share your opinion.

Reply:
I think we first need to confirmation we are breaking compatibility with this issue and PR. We are renaming an interface that sits in the API class. I am also not seeing any deprecation in this PR as issue states.

If we are breaking compatibility, then issues needs label.

Reply:
we put labels after a fix, if we agree on fix, label is just reflection.
I am ok to break compatibility if we got better api that is easier to understand.